File: Java/convert1DArrayInto2DArray/Solution.java
// https://leetcode.com/problems/convert-1d-array-into-2d-array/

package convert1DArrayInto2DArray;

class Solution {

    public int[][] construct2DArray(int[] original, int row, int col) {
		if (row * col != original.length)
			return new int[0][0];

		int[][] arr = new int[row][col];
		int idx = 0;
		for (int i = 0; i < arr.length; ++i) {
			for (int j = 0; j < arr[i].length; ++j)
				arr[i][j] = original[idx++];
		}

		return arr;
    }
}

File: cpp/convert1DArrayInto2DArray/solution.cpp
// https://leetcode.com/problems/convert-1d-array-into-2d-array/

#include <iostream>
#include <vector>

using namespace std;

class Solution
{
	public:
		vector<vector<int>> construct2DArray(vector<int>& original, int row, int col)
		{
			vector<vector<int>> v;
			if (row * col != original.size())
				return v;

			for (auto n : original) {
				if (v.empty() || v.back().size() == col)
					v.push_back({});
				v.back().push_back(n);
			}
			return v;
		}
};

int main(int argc, char** argv)
{
}
